adapters/repos/db/shard_write_batch_delete.go

```diff
@@ -0,0 +1,142 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package db
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/semi-technologies/weaviate/adapters/repos/db/inverted"
+	"github.com/semi-technologies/weaviate/entities/additional"
+	"github.com/semi-technologies/weaviate/entities/filters"
+	"github.com/semi-technologies/weaviate/entities/storagestate"
+	"github.com/semi-technologies/weaviate/usecases/objects"
+)
+
+// return value map[int]error gives the error for the index as it received it
+func (s *Shard) deleteObjectBatch(ctx context.Context,
+	docIDs []uint64, dryRun bool) objects.BatchSimpleObjects {
+	if s.isReadOnly() {
+		return objects.BatchSimpleObjects{
+			objects.BatchSimpleObject{Err: storagestate.ErrStatusReadOnly},
+		}
+	}
+	return newDeleteObjectsBatcher(s).Delete(ctx, docIDs, dryRun)
+}
+
+type deleteObjectsBatcher struct {
+	sync.Mutex
+	shard   *Shard
+	objects objects.BatchSimpleObjects
+}
+
+func newDeleteObjectsBatcher(shard *Shard) *deleteObjectsBatcher {
+	return &deleteObjectsBatcher{shard: shard}
+}
+
+func (b *deleteObjectsBatcher) Delete(ctx context.Context,
+	docIDs []uint64, dryRun bool) objects.BatchSimpleObjects {
+	b.delete(ctx, docIDs, dryRun)
+	b.flushWALs(ctx)
+	return b.objects
+}
+
+func (b *deleteObjectsBatcher) delete(ctx context.Context,
+	docIDs []uint64, dryRun bool) {
+	b.objects = b.deleteSingleBatchInLSM(ctx, docIDs, dryRun)
+}
+
+func (b *deleteObjectsBatcher) deleteSingleBatchInLSM(ctx context.Context,
+	batch []uint64, dryRun bool) objects.BatchSimpleObjects {
+	result := make(objects.BatchSimpleObjects, len(batch))
+	objLock := &sync.Mutex{}
```

Comment:
does the Mutex need to be addressable?

Reply:
that one I must say that it's copied from the batch create objects method :) so I would leave it :)